File: ScorpioProto/Java/src/ScorpioProto/Commons/IScorpioReader.java
package ScorpioProto.Commons;

import java.util.Calendar;

public interface IScorpioReader {
	boolean ReadBool();
	byte ReadInt8();
	byte ReadUInt8();
	short ReadInt16();
	short ReadUInt16();
	int ReadInt32();
	int ReadUInt32();
	long ReadInt64();
	long ReadUInt64();
	float ReadFloat();
	double ReadDouble();
	String ReadString();
	Calendar ReadDateTime();
	byte[] ReadBytes();
}

File: ScorpioProto/CSharp/ScorpioProto/src/ScorpioProto/Commons/IScorpioReader.cs
using System;
namespace ScorpioProto.Commons {
    public interface IScorpioReader : IDisposable {
        int ReadHead(string fileName, string MD5);
        bool ReadBool();
        sbyte ReadInt8();
        byte ReadUInt8();
        short ReadInt16();
        ushort ReadUInt16();
        int ReadInt32();
        uint ReadUInt32();
        long ReadInt64();
        ulong ReadUInt64();
        float ReadFloat();
        double ReadDouble();
        string ReadString();
        DateTime ReadDateTime();
        byte[] ReadBytes();
    }
}